src/setup-beam.js

```diff
@@ -17,21 +18,49 @@ async function main() {
   const elixirSpec = core.getInput('elixir-version', { required: false })
   const gleamSpec = core.getInput('gleam-version', { required: false })
   const rebar3Spec = core.getInput('rebar3-version', { required: false })
+  const shouldMixRebar = core.getInput('install-rebar', { required: false })
+  const shouldMixHex = core.getInput('install-hex', { required: false })
+  process.env.HEX_MIRROR = core.getInput('hexpm-mirror', { required: false })
 
   if (otpSpec !== 'false') {
-    await installOTP(otpSpec, osVersion)
-    const elixirInstalled = await maybeInstallElixir(elixirSpec, otpSpec)
+    const otpVersion = await getOTPVersion(otpSpec, osVersion)
+    const elixirVersion = await getElixirVersion(elixirSpec, otpVersion)
+    // No cache for Windows OTP, rebar3 and Gleam as they're downloaded from GitHub
+    const useCache =
+      isVersion(otpSpec) &&
+      (!elixirSpec || isVersion(elixirSpec)) &&
+      (elixirVersion || process.platform !== 'win32')
+    let hexCachedRestored = false
+    const cacheVersion = '1'
+    const hexCacheKey =
+      `hex.pm-${osVersion}-${otpVersion}-${elixirVersion}-` +
+      `r=${shouldMixRebar}-h=${shouldMixHex}-${cacheVersion}`
+    const hexCachePaths = [
+      path.join(process.env.RUNNER_TEMP, '.setup-beam', 'cache'),
+      path.join(
```

Comment:
If we're not caching on Windows is this required?

Reply:
Sorry I am confused about what would not be required. Do you mean the call to the path API instead of directly using POSIX-style paths here?

This may be unrelated, but we do cache on Windows, just not Erlang/OTP.

Reply:
Ah, I must've been confused by a previous comment. I though we weren't caching on Windows at all, but if it's just OTP, it's Ok to keep this. Resolving...